apps/tlon-mobile/src/hooks/useNotificationListener.ts

```diff
@@ -65,12 +72,51 @@ function payloadFromNotification(
 
   // welcome to my validation library ;)
   if (payload.wer != null && payload.channelId != null) {
+    const postInfo = api.getPostInfoFromWer(payload.wer);
+    const handoffPost: db.Post | undefined = (() => {
+      const dmPost = payload.dmPost as ub.DmPostEvent['dm-post'] | undefined;
+      if (dmPost != null) {
+        // key.id looks like `dm/000.000.000.mor.eme.ssa.gei.dxx`
+        const id = dmPost.key.id.split('/')[1];
+        const receivedAt = getReceivedAtFromId(id);
+        return {
```

Comment:
would definitely like a second look here to ensure I'm connecting the API resource to db.Post correctly

Reply:
Looks right to my eye -- would it make sense to move this logic to modelBuilders?

Reply:
done! [9ad3193](https://github.com/tloncorp/tlon-apps/pull/4280/commits/9ad31934220b8183f240003a6eab9cae44c9665c)